### PR TITLE
 IAM BitbucketFileServerPolicy updated

### PR DESCRIPTION
**Pull request by jjeongatl**
BitbucketFileServerPolicy fields updated based on provisioned resources.

### CODE REVIEW

File: .cfnlintrc [hunk 1]
```diff
@@ -1,5 +1,5 @@
 templates:
-  - templates/*
+  - /*
```
Comment (nanux) [inline .cfnlintrc]: Why this change?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+repos:
+- repo: https://github.com/aws-cloudformation/cfn-python-lint
+  rev: v0.53.0  # The version of cfn-lint to use
+  hooks:
+    - id: cfn-python-lint
+      # files: templates/.*\.(json|yml|yaml)$
```
Comment (nanux) [inline .pre-commit-config.yaml]: I believe we can leave it uncommented

File: templates/quickstart-bitbucket-dc.template.yaml [hunk 3]
```diff
@@ -729,32 +729,45 @@ Resources:
...
+                  - ec2:CreateSnapshot
+                  - rds:DescribeDBInstances
+                  - rds:ModifyDBInstance
                 Resource:
-                  - !Sub "arn:${AWS::Partition}:ec2:*:${AWS::AccountId}:instance/*"
-                  - !Sub "arn:${AWS::Partition}:ec2:*:${AWS::AccountId}:volume/*"
+                  - arn:aws:rds:*:887764444972:snapshot:*
+                  - arn:aws:ec2:*:887764444972:instance/*
+                  - arn:aws:ec2:*:887764444972:volume/*
+                  - arn:aws:autoscaling:*:887764444972:autoScalingGroup:*:autoScalingGroupName/*
```
Comment (nanux) [inline templates/quickstart-bitbucket-dc.template.yaml]: We cannot include specific AWS account (`887764444972`) in the general AWS quickstart.

Can we change the account to `*` as well?

File: templates/quickstart-bitbucket-dc.template.yaml [hunk 4]
```diff
@@ -765,27 +778,15 @@ Resources:
...
-                  - rds:CopyDBSnapshot
-                  - rds:DescribeDBInstances
-                  - rds:ModifyDBInstance
-                  - rds:DeleteDBSnapshot
-                  - rds:PromoteReadReplica
-                  - iam:PassRole
-                  - autoscaling:CreateOrUpdateTags
-                  - es:*
+              - Action: iam:PassRole
+                Resource: arn:aws:iam::887764444972:role/*
```
Comment (nanux) [inline templates/quickstart-bitbucket-dc.template.yaml]: Remove the account

File: templates/quickstart-bitbucket-dc.template.yaml [hunk 5]
```diff
@@ -765,27 +778,15 @@ Resources:
...
+                Effect: Allow
+              - Action: es:*
                 Resource:
-                  - !Sub "arn:${AWS::Partition}:iam::${AWS::AccountId}:role/*"
-                  - !Sub "arn:${AWS::Partition}:es:*:${AWS::AccountId}:domain/*"
-                  - !Sub "arn:${AWS::Partition}:autoscaling:*:${AWS::AccountId}:autoScalingGroup:*:autoScalingGroupName/*"
-                  - !Sub "arn:${AWS::Partition}:rds:*:${AWS::AccountId}:snapshot:*"
+                  - arn:aws:es:*:887764444972:pipeline/*
+                  - arn:aws:es:*:887764444972:domain/*
```
Comment (nanux) [inline templates/quickstart-bitbucket-dc.template.yaml]: Again, the account

**Review by nanux — CHANGES_REQUESTED** [review]

File: .cfnlintrc (hunk 1 above)
Comment (jjeongatl) [reply .cfnlintrc]: reverted

File: templates/quickstart-bitbucket-dc.template.yaml (hunk 3 above)
Comment (jjeongatl) [reply templates/quickstart-bitbucket-dc.template.yaml]: resolved

File: templates/quickstart-bitbucket-dc.template.yaml (hunk 5 above)
Comment (jjeongatl) [reply templates/quickstart-bitbucket-dc.template.yaml]: resolved

**Comment by jjeongatl** [comment]
Waiting on aws reply regarding ERDSDBInstancePubliclyAccessible & EIAMPolicyWildcardResource exclusion

File: templates/quickstart-bitbucket-dc.template.yaml [hunk 6]
```diff
@@ -1079,7 +1156,83 @@ Resources:
           - Effect: "Allow"
             Principal:
               AWS: !GetAtt BitbucketClusterNodeRole.Arn
-            Action: "es:*"
+            Action:
```
Comment (nanux) [inline templates/quickstart-bitbucket-dc.template.yaml:1161]: Can we sort this alphabetically?

Comment (errcode1202) [reply templates/quickstart-bitbucket-dc.template.yaml:1159]: 💯

File: templates/quickstart-bitbucket-dc.template.yaml [hunk 7]
```diff
@@ -765,27 +778,91 @@ Resources:
...
+                  - es:RejectInboundCrossClusterSearchConnection
+                  - es:ESCrossClusterGet
+                  - es:UpdateDomainConfig
+                  - es:StartElasticsearchServiceSoftwareUpdate
+                  - es:DescribeDomains
+                  - es:ListInstanceTypeDetails
+                  - es:RemoveTags
+                  - es:ListElasticsearchVersions
+                  - es:AssociatePackage
+                  - es:DescribeElasticsearchDomain
```
Comment (errcode1202) [inline templates/quickstart-bitbucket-dc.template.yaml]: Going to assume our CI plans are all passing with this?

File: templates/quickstart-bitbucket-dc.template.yaml [hunk 8]
```diff
@@ -765,27 +778,91 @@ Resources:
...
-                  - rds:CreateDBSnapshot
-                  - rds:RestoreDBInstanceFromDBSnapshot
-                  - rds:CopyDBSnapshot
-                  - rds:DescribeDBInstances
-                  - rds:ModifyDBInstance
-                  - rds:DeleteDBSnapshot
-                  - rds:PromoteReadReplica
-                  - iam:PassRole
-                  - autoscaling:CreateOrUpdateTags
-                  - es:*
```
Comment (errcode1202) [inline templates/quickstart-bitbucket-dc.template.yaml:785]: @nanux do we think these changes are backward compatible?

File: templates/quickstart-bitbucket-dc.template.yaml (hunk 7 above)
Comment (jjeongatl) [reply templates/quickstart-bitbucket-dc.template.yaml]: Yes

File: templates/quickstart-bitbucket-dc.template.yaml (hunk 8 above)
Comment (jjeongatl) [reply templates/quickstart-bitbucket-dc.template.yaml:785]: @errcode1202 @nanux Any comments on this?

**Review by nghazali — APPROVED** [review]
